src/wallet/wallet.cpp

```diff
@@ -166,11 +166,36 @@ bool CWallet::AddSaplingZKey(
         return true;
     }
 
-    // TODO: Persist to disk
+    if (!IsCrypted()) {
+        auto ivk = sk.expsk.full_viewing_key().in_viewing_key();
+        return CWalletDB(strWalletFile).WriteSaplingZKey(ivk, sk, mapSaplingZKeyMetadata[ivk]);
```

Comment:
`mapSaplingZKeyMetadata` has type `std::map<libzcash::SaplingPaymentAddress, CKeyMetadata>`, and if the key doesn't exist in the map, `std::map`'s `operator[]` uses the default constructor of the target type to insert a new instance. In this case it will be a `CKeyMetadata` instance with the current version and an "unknown" time.

I'm unsure whether that was intended or whether this is a case that shouldn't ever happen.

Reply:
This is, I believe, expected behaviour:

- When calling `AddSaplingZKey` from `GenerateNewSaplingZKey`, we set `mapSaplingZKeyMetadata` there first.
- When calling `AddSaplingZKey` from `AddSpendingKeyToWallet`, we set `mapSaplingZKeyMetadata` appropriately afterwards.